python/GafferImageUI/ImageViewUI.py

```diff
@@ -275,80 +298,193 @@ def _hsvString( color ) :
+			self.__hsvLabel = GafferUI.Label()
+			self.__hsvLabel._qtWidget().setFixedWidth( labelFontMetrics.width( "HSV : 99999 99999 99999" ) )
+
+			self.__exposureLabel = GafferUI.Label()
+			self.__exposureLabel._qtWidget().setFixedWidth( labelFontMetrics.width( "EV : 19.9" ) )
+
+			if mode == 2:
+				m = IECore.MenuDefinition()
+				m.append( "/Pixel Inspector",
+					{ "command" : functools.partial( Gaffer.WeakMethod( self.__addClick ), False ) }
+				)
+				m.append( "/Region Inspector",
+					{ "command" : functools.partial( Gaffer.WeakMethod( self.__addClick ), True ) }
+				)
+				button = GafferUI.MenuButton( "", "plus.png", hasFrame=False, menu = GafferUI.Menu( m, title = "Add Color Inspector" ) )
+			else:
+				button = GafferUI.Button( "", "delete.png", hasFrame=False )
+				button.clickedSignal().connect( Gaffer.WeakMethod( self.__deleteClick ), scoped = False )
+
+
+		self.__pixel = imath.V2i( 0 )
+
+
+		if plug.getName() == "ColorInspectorPlug":
+			viewportGadget = plug.node().viewportGadget()
+			viewportGadget.mouseMoveSignal().connect( Gaffer.WeakMethod( self.__mouseMove ), scoped = False )
+
+			imageGadget = viewportGadget.getPrimaryChild()
+			imageGadget.buttonPressSignal().connect( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+			imageGadget.dragBeginSignal().connect( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+			imageGadget.dragEnterSignal().connect( Gaffer.WeakMethod( self.__dragEnter ), scoped = False )
+			imageGadget.dragMoveSignal().connect( Gaffer.WeakMethod( self.__dragMove ), scoped = False )
+			imageGadget.dragEndSignal().connect( Gaffer.WeakMethod( self.__dragEnd ), scoped = False )
+
+		self.__swatch.buttonPressSignal().connect( Gaffer.WeakMethod( self.__buttonPress ), scoped = False )
+		self.__swatch.dragBeginSignal().connect( Gaffer.WeakMethod( self.__dragBegin ), scoped = False )
+		self.__swatch.dragEndSignal().connect( Gaffer.WeakMethod( self.__dragEnd ), scoped = False )
+
+		plug.node()["colorInspector"]["evaluator"]["pixelColor"].getInput().node().plugDirtiedSignal().connect( Gaffer.WeakMethod( self._updateFromImageNode ), scoped = False )
+
+		plug.node().plugDirtiedSignal().connect( Gaffer.WeakMethod( self._plugDirtied ), scoped = False )
+		plug.node()["in"].getInput().node().scriptNode().context().changedSignal().connect( Gaffer.WeakMethod( self._updateFromContext ), scoped = False )
+		Gaffer.Metadata.plugValueChangedSignal( self.getPlug().node() ).connect( Gaffer.WeakMethod( self.__plugMetadataChanged ), scoped = False )
 
 		self.__updateLabels( imath.V2i( 0 ), imath.Color4f( 0, 0, 0, 1 ) )
 
+		# Set initial state of mode icon
+		self._plugDirtied( plug["mode"] )
+
+	def addInspector( self ):
+		parent = self.getPlug().parent()
+		suffix = 1
+		while "c" + str( suffix ) in parent:
+			suffix += 1
+
+		parent.addChild( GafferImageUI.ImageView.ColorInspectorPlug( "c" + str( suffix ) ) )
```

Comment:
This felt more desirable, with deletions not changing all the labels, and insertions filling in gaps ( using addChild's automated version, new suffixes are always greater than the max suffix, instead of filling in gaps ( ie.  add 1 - 10, remove 1-9 leaving just 10, does the next insert readd 1, or add 11? ).

As mentioned above, if you feel strongly about this, we can probably sacrifice a bit of usability.